src/orchestratord/src/controller/materialize/environmentd.rs

```diff
@@ -1681,7 +1693,11 @@ fn create_connection_info(
             },
         },
     };
-    if authenticator_kind == AuthenticatorKind::Password {
+
+    if matches!(
+        auth_kind,
+        AuthenticatorKind::Password | AuthenticatorKind::Sasl
+    ) {
```

Comment:
If we are checking this a bunch perhaps at some point we could add a `fn authenticated(&self) -> bool` to AuthenticatorKind, but fine for now

Reply:
Yeah good call! Especially once we start adding SSO/JWT auth, having something like `authenticated()` will probably make things cleaner. Worth keeping in mind.